addons/account_peppol/wizard/account_move_send.py

```diff
@@ -123,6 +126,10 @@ def action_send_and_print(self, force_synchronous=False, allow_fallback_pdf=Fals
         # Extends 'account' to force ubl xml checkbox if sending via peppol
         self.ensure_one()
 
+        can_send = self.env['account_edi_proxy_client.user']._get_can_send_domain()
+        if self.checkbox_send_peppol and self.company_id.account_peppol_proxy_state not in can_send:
+            return self.env['peppol.registration'].with_company(self.company_id)._action_open_peppol_form(reopen=False)
```

Comment:
It's weird you need `with_company` here. Create the wizard directly with the correct company instead